lib/init.js

```diff
@@ -0,0 +1,99 @@
+'use babel';
+
+let bailey;
+let helpers;
+
+const levelMapping = {
+  SyntaxError: 'error',
+  ImportError: 'error',
+  StyleError: 'warning',
+};
+
+const formatMessage = (error) => {
+  if (error.found) return `Found unexpected ${error.inner}`;
+  return error.message;
+};
+
+const loadDeps = () => {
+  if (!bailey) {
+    bailey = require('bailey');
+  }
+  if (!helpers) {
+    helpers = require('atom-linter');
+  }
+};
+
+export default {
+  activate() {
+    this.idleCallbacks = new Set();
+    let depsCallbackID;
+    const installLinterBaileyDeps = () => {
+      this.idleCallbacks.delete(depsCallbackID);
+      if (!atom.inSpecMode()) {
+        require('atom-package-deps').install('linter-bailey');
+      }
+      loadDeps();
+      if (atom.inDevMode()) {
+        // eslint-disable-next-line no-console
+        console.log('linter-bailey: All dependencies installed.');
+      }
+    };
+    depsCallbackID = window.requestIdleCallback(installLinterBaileyDeps);
+    this.idleCallbacks.add(depsCallbackID);
+
+    if (atom.inDevMode()) {
+      /* eslint-disable no-console */
+      console.log('linter-bailey: bailey linter is now activated.');
+      /* eslint-enable no-console */
+    }
+  },
+
+  deactivate() {
+    this.idleCallbacks.forEach(callbackID => window.cancelIdleCallback(callbackID));
+    this.idleCallbacks.clear();
+  },
+
+  provideLinter() {
+    return {
+      name: 'bailey',
+      grammarScopes: [
+        'source.bs',
+      ],
+      scope: 'file',
+      lintsOnChange: true,
+      lint: async (editor) => {
+        const filePath = editor.getPath();
+        if (!filePath) {
+          // Somehow a TextEditor without a path was passed in
+          return null;
+        }
+
+        const fileText = editor.getText();
+        loadDeps();
+
+        const messages = [];
+        try {
+          bailey.parseString(fileText);
+        } catch (e) {
+          if (e instanceof bailey.ParserError) {
+            messages.push({
+              severity: levelMapping[e.inner.name],
```

Comment:
What happens if it's not one of the three mappings you have defined? The `Set` or other approaches you have used elsewhere should work fine here.